spec/controllers/version_controller_spec.cr

```diff
@@ -0,0 +1,84 @@
+require "./spec_helper"
+
+def product_hash
+  {"name" => "Fake"}
+end
+
+def create_product
+  model = Product.new(product_hash)
+  model.save
+  model
+end
+
+def version_hash
+  {
+    "product_id" => create_product.id.to_s,
+    "image" => "00.XXXXXXX"
+  }
+end
+
+def version_params
+  params = [] of String
+  params << "image=#{version_hash["image"]}"
+  params << "product_id=#{version_hash["product_id"]}"
+  params.join("&")
+end
+
+def invalid_version_params
+  params = [] of String
+  params << "image="
+  params.join("&")
+end
+
+def create_version
+  version = Version.new(version_hash)
+  version.save
+  version
+end
+
+
+class VersionControllerTest < GarnetSpec::Controller::Test
+  getter handler : Amber::Pipe::Pipeline
+
+  def initialize
+    @handler = Amber::Pipe::Pipeline.new
+    @handler.build :web do
+      plug Amber::Pipe::Error.new
+      plug Amber::Pipe::Session.new
+      plug Amber::Pipe::Flash.new
+    end
+    @handler.prepare_pipelines
+  end
+end
+
+describe VersionController do
+  subject = VersionControllerTest.new
+
+  describe "VersionController#index" do
+    it "renders the view" do
+      Version.clear
```

Comment:
Version.clear isn't necessary because granite clear the database before each test